src/compiler/checker.ts

```diff
@@ -4583,7 +4583,22 @@ namespace ts {
                 let requireOptionalProperties = relation === subtypeRelation && !(source.flags & TypeFlags.ObjectLiteral);
                 for (let targetProp of properties) {
                     let sourceProp = getPropertyOfType(source, targetProp.name);
-                    if (sourceProp !== targetProp) {
+
+                    if (sourceProp === targetProp) { // source inherits targetProp and doesn't redeclare/override it.

```

Comment:
I also changed the error message, cf d0924f43f031d95447822f2e74651fca2529115a. Which is better?